python/pyiceberg/io/pyarrow.py

```diff
@@ -223,8 +236,8 @@ def new_input(self, location: str) -> PyArrowFile:
         Returns:
             PyArrowFile: A PyArrowFile instance for the given location
         """
-        fs, path = self.get_fs_and_path(location)
-        return PyArrowFile(fs=fs, location=location, path=path)
+        fs = self.get_fs(self.extract_scheme(location))
+        return PyArrowFile(fs=fs, location=location)
```

Comment:
I don't understand why this parses the URI here and in `PyArrowFile`. I think this should just leave PyArrowFile the way that it was and pass in the path after parsing it here:

```python
    scheme, path = parse_location(location)
    fs = self.get_fs(scheme)
    return PyArrowFile(fs=fs, location=location, path=path)
```

Reply:
My main argument here is, that if you want to use the PyArrowFile on its own, then the user is responsible for setting the location and path, which I think should be kept internal of the object.